### ISSUE TITLE
Fatal error when connecting over TLS occures randomly

### ISSUE THREAD

**Issue by szaqal83** [DirectoryTree/LdapRecord#58]
LDAP Server Type: **Active Directory**
PHP Version: **7.4.1**

Description:
"Fatal error: Uncaught ErrorException: ldap_start_tls(): Unable to start TLS..." error occures randomly. I get this error almost every second execution of script. No modifications on server or script are done. Some time it connects over TLS without any problems, sometime it doesn't, throwing TLS connection error.

Config:
`$config = array(
	'hosts' => array('10.10.50.25'),
	'base_dn' => 'dc=XXX,dc=ZZ',
	'username' => 'user',
	'password' => 'pass',
	// Optional Configuration Options
	'use_tls' => true,
	// Custom LDAP Options
	'options' => array(
		// See: http://php.net/ldap_set_option
		LDAP_OPT_X_TLS_REQUIRE_CERT => LDAP_OPT_X_TLS_ALLOW
    )
);`

Steps To Reproduce:
Execute script.

**Comment by szaqal83** [comment]
I've made some more tests, switched to PHP 7.3 and no more error occurrence. It seems to be PHP related bug.

**Comment by stevebauman** [comment]
Hi @szaqal83, that’s definitely strange...

I have been doing all of my own testing in PHP 7.3. I’ll upgrade to 7.4 and see if I encounter the same bug.

**Comment by stevebauman** [comment]
Closed wrong issue - apologies.

**Comment by lumpygator** [comment]
I had a similar issue today. After every successful request, the next request was throwing the same fatal error.

I downgraded my Docker image from PHP 7.3.12-fpm to 7.3.11-fpm and the issue was resolved.  I then upgraded back to 7.3.12 expecting to reproduce the same issue but could no longer reproduce it.

After reviewing my notes, I had updated the `/etc/ldap/ldap.conf` file in my Docker PHP container to not require TLS cert validation (i.e. `TLS_REQCERT never`) but never restarted the Docker container.

I'm not sure if you did something similar, but it seems that updating the client LDAP conf without restarting PHP might have been the issue.  

I've yet to reproduce since re-building and then restarting after making changes to the LDAP client conf.
